app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -384,19 +378,25 @@ class HomeFragment : Fragment() {
             )
         )
 
-        requireComponents.backgroundServices.accountManager.register(currentMode, owner = this)
-        requireComponents.backgroundServices.accountManager.register(object : AccountObserver {
-            override fun onAuthenticated(account: OAuthAccount, authType: AuthType) {
-                if (authType != AuthType.Existing) {
-                    view?.let {
-                        FenixSnackbar.make(it, Snackbar.LENGTH_SHORT)
-                            .setText(it.context.getString(R.string.onboarding_firefox_account_sync_is_on))
-                            .setAnchorView(toolbarLayout)
-                            .show()
+        requireComponents.backgroundServices.accountManagerAvailableQueue.runIfReadyOrQueue {
+            // By the time this code runs, we may not be attached to a context.
+            if ((this@HomeFragment).context == null) {
```

Comment:
We could use https://github.com/mozilla-mobile/fenix/blob/be7bd527f68a4a128be5f3f27f93650da0fc6f3e/app/src/main/java/org/mozilla/fenix/addons/Extensions.kt#L39

Reply:
or we could combine the two.